### PR TITLE
[orders api] Created the orders api test app in php

### PR DESCRIPTION
**Pull request by mayankamencherla**
__DESCRIPTION__

- [x] Moved old code to automatic checkout folder
- [x] Added a new folder called orders api which contains a sample app for orders api
- [x] Add latest PHP SDK and update code
- [x] Add a cool UI and some HTML / CSS
- [x] Code review

### CODE REVIEW

File: automatic/index.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <form action="charge.php" method="POST">
       <script
         src="https://checkout.razorpay.com/v1/checkout.js"
-        data-key="rzp_test_WyK93y9mvps7SN"
+        data-key="<Your Key Id>"
```
Comment (captn3m0) [inline automatic/index.html]: 1. Move this to `config.php` as discussed
2. Maybe use `index.php?checkout=automatic|manual`

File: demo.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<html>
+<head lang="en">
+    <meta charset="utf-8">
+</head>
+<body>
+    <form id="checkout-selection" method="POST">
+        <input type="radio" name="checkout" value="automatic">Automatic Checkout Demo<br>
```
Comment (captn3m0) [inline demo.html]: This is just too much complication where 2 anchor tags would work equally well.

File: manual/order.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+$html = <<<EOT
+<button id="rzp-button1">Pay with Razorpay</button>
+<script src="https://checkout.razorpay.com/v1/checkout.js"></script>
+<form name='razorpayform' action="verify-signature.php" method="POST">
+    <input type="hidden" name="razorpay_payment_id" id="razorpay_payment_id">
+    <input type="hidden" name="razorpay_signature"  id="razorpay_signature" >
+</form>
+<script>
+var options = {
```
Comment (captn3m0) [inline manual/order.php]: We want to support both automatic and manual checkout but only with orders API. The automatic/index.html is still on old integration, and the new one is only on manual checkout.

File: manual/order.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    "key": "$keyId",
+    "amount": "$amount", 
+    "name": "Merchant Name",
+    "description": "Purchase Description",
+    "image": "/your_logo.png",
```
Comment (captn3m0) [inline manual/order.php]: Put a HTTPS link.

Also, go through the entire checkout-fields list and add as many comments and example values here as possible (`modal.ondismiss` for eg)

File: automatic/index.html (hunk 1 above)
Comment (mayankamencherla) [inline automatic/index.html]: Moved to config.php

File: demo.html (hunk 2 above)
Comment (mayankamencherla) [inline demo.html]: I agree. But for the future I would like to design the app a lot better, would make for a cool test app. Adding radio buttons would leave things open ended for the future.

File: manual/order.php (hunk 3 above)
Comment (mayankamencherla) [inline manual/order.php]: Changed to orders api

File: manual/order.php (hunk 4 above)
Comment (mayankamencherla) [inline manual/order.php]: added. But haven't added comments and example values yet. Will add them later.

**Comment by mayankamencherla** [comment]
@captn3m0 Will wait for the new php sdk to go out. It has a new `verifyPaymentSignature` method.

File: checkout/automatic.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<!--  The entire list of Checkout fields is available at
+ https://docs.razorpay.com/docs/checkout-form#checkout-fields -->
+
+<form action="verify.php" method="POST">
+  <script
+    src="https://checkout.razorpay.com/v1/checkout.js"
+    data-key="<?php echo $keyId?>"
```
Comment (captn3m0) [inline checkout/automatic.php]: Pick up data here from what was passed in $data instead

File: pay.php [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+// Create the Razorpay Order
+
+use Razorpay\Api\Api;
+
+$api = new Api($keyId, $keySecret);
+
+$orderData = [
+    'receipt'         => 3456,
+    'amount'          => 2000 * 100, // 2000 rupees in paise
```
Comment (captn3m0) [inline pay.php]: Have a variable for $amount instead, so it is easier to understand?
receipt -> explain what the variable does?

leave as many comments as you can...

File: checkout/automatic.php [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    data-amount="<?php echo $amount?>"
+    data-currency="INR"
+    data-name="Daft Punk"
+    data-description="Purchase Description"
+    data-image="../daft-punk.jpg"
```
Comment (captn3m0) [inline checkout/automatic.php]: Upload the file to imgur instead and use that (https link)

Also comment that the URL needs to be HTTPS to work

Comment (captn3m0) [inline checkout/automatic.php]: Also comment about the dashboard upload feature

File: pay.php [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    'currency'        => 'INR',
+    'payment_capture' => 1 // auto capture
```
Comment (captn3m0) [inline pay.php]: leave a link here to the orders API docs so people can figure out what this means

File: verify.php [hunk 9]
```diff
@@ -0,0 +1,50 @@
+<?php
+
+require('config.php');
+
+//These should be commented out in production
+error_reporting(E_ALL);
```
Comment (captn3m0) [inline verify.php]: Drop error reporting... Or maybe do it via a config flag?

File: verify.php [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+        // could be database or something else)
+        $attributes = array(
+            'razorpay_order_id' => $_SESSION['razorpay_order_id'],
+            'razorpay_payment_id' => $_POST['razorpay_payment_id'],
+            'razorpay_signature' => $_POST['razorpay_signature']
+        );
+
+        $success = $api->utility->verifyPaymentSignature($attributes);
+    }
+    catch(Exception $e)
```
Comment (captn3m0) [inline verify.php]: Wait for https://github.com/razorpay/razorpay-php/pull/27 to be merged and then drop the try catch. verifySignature should not be throwing exceptions

File: verify.php [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    $html = "<p>Your payment was successful</p>
+             <p>Payment ID: {$_POST['razorpay_payment_id']}</p>";
+}
+else 
+{
+    $html = "<p>Your payment failed</p>
+             <p>Error Message: Payment Failed</p>";
+}
+
+echo $html;
```
Comment (captn3m0) [inline verify.php]: Add a editorconfig file, so we get a newline at the end

File: checkout/automatic.php (hunk 7 above)
Comment (captn3m0) [inline checkout/automatic.php]: still using local image link. We need to pick as many values as we can from the data that was passed.

File: verify.php (hunk 10 above)
Comment (captn3m0) [inline verify.php]: Use the exact exception we are throwing

File: verify.php (hunk 11 above)
Comment (captn3m0) [inline verify.php]: You are giving two different error messages. Move the success conditionals inside either the try or the catch instead and give a single error response.

**Pull request merged**